tests/integration/test_host_group.py

```diff
@@ -144,3 +149,86 @@ async def _test_pagination_action_host_group(host_group_node: ActionHostGroupNod
 
     assert len(await host_group_node.all()) == 54
     assert len(await host_group_node.filter()) == 54
+
+
+# test_change_hosts_by_filter
+
+
+async def test_change_hosts_by_filter(adcm_client: ADCMClient, cluster: Cluster, hostprovider: HostProvider) -> None:
+    h1, h2, h3 = await asyncio.gather(*(adcm_client.hosts.create(hostprovider, f"host-{i}") for i in range(3)))
+    await cluster.hosts.add((h1, h2))
+    service, *_ = await cluster.services.add(Filter(attr="name", op="eq", value="example_1"))
+    component, *_ = await service.components.all()
+    mapping = await cluster.mapping
+    await mapping.add(component, h1)
+    await mapping.save()
+
+    # hostprovider
+    group = await hostprovider.config_host_groups.create(name="chg")
+    assert len(await group.hosts.all()) == 0
+    await group.hosts.add(Filter(attr="name", op="eq", value=h3.name))
+    assert len(await group.hosts.all()) == 1
+    await group.hosts.add(Filter(attr="name", op="contains", value="host"))
+    assert len(await group.hosts.all()) == 3
+    await group.hosts.remove(Filter(attr="name", op="in", value=(h1.name, h2.name)))
+    assert len(await group.hosts.all()) == 1
+
+    await group.hosts.set(Filter(attr="name", op="in", value=(h1.name, h2.name)))
```

Comment:
For what reason did you not add `set()` method checks for the host groups of cluster, service, and component objects?

Reply:
only for test simplicity